ksql-functional-tests/src/test/java/io/confluent/ksql/test/planned/TestCasePlan.java

```diff
@@ -16,55 +16,80 @@
 package io.confluent.ksql.test.planned;
 
 import io.confluent.ksql.engine.KsqlPlan;
+import io.confluent.ksql.test.model.RecordNode;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 
 public final class TestCasePlan {
-  private final TestCasePlanNode node;
+  private final TestCaseSpecNode specNode;
+  private final TestCasePlanNode planNode;
   private final String topology;
 
   TestCasePlan(
       final String version,
       final long timestamp,
-      final List<KsqlPlan> plan,
+      final List<KsqlPlan> planNode,
       final String topology,
       final Map<String, String> schemas,
-      final Map<String, String> configs
+      final Map<String, String> configs,
+      final List<RecordNode> inputs,
+      final List<RecordNode> outputs
   ) {
-    this(new TestCasePlanNode(version, timestamp, plan, schemas, configs), topology);
+    this(
+        new TestCaseSpecNode(version, timestamp, schemas, inputs, outputs),
+        new TestCasePlanNode(planNode, configs),
+        topology
+    );
   }
 
-  TestCasePlan(final TestCasePlanNode node, final String topology) {
-    this.node = Objects.requireNonNull(node, "node");
+  TestCasePlan(
+      final TestCaseSpecNode specNode,
+      final TestCasePlanNode planNode,
+      final String topology
+  ) {
+    this.specNode = Objects.requireNonNull(specNode, "spec");
+    this.planNode = Objects.requireNonNull(planNode, "plan");
     this.topology = Objects.requireNonNull(topology, "topology");
   }
 
   public List<KsqlPlan> getPlan() {
-    return node.getPlan();
+    return planNode.getPlan();
```

Comment:
I'd inline all of these accessors that access either planNode or specNode. Calls can just use `getPlanNode().getPlan()` etc.